### ISSUE TITLE
`no-redeclare` should not check `let` and `const`

### ISSUE THREAD

**Issue by magurotuna** [denoland/deno_lint#772]
### Code Snippet

```ts
let a = 1;
let a = 2;

const b = 3;
const b = 4;
```

### Expected Result

No `no-redeclare` errors

### Actual Result

```
❯ cargo run --quiet --example dlint -- run --rule no-redeclare maguro.ts
error[no-redeclare]: Redeclaration is not allowed
 --> maguro.ts:2:5
  |
2 | let a = 2;
  |     ^
  |
error[no-redeclare]: Redeclaration is not allowed
 --> maguro.ts:5:7
  |
5 | const b = 4;
  |       ^
  |
Found 2 problems
```

### Additional Info

In fact `no-redeclare` and `@typescript-eslint/no-redeclare` don't check `let` and `const`. These two declaration kinds, if redeclared, are just syntax errors, so it's not an error the linter should cover.

### Version

34e4147

**Comment by lawrence-dol** [comment]
I have to disagree; whether a syntax error or not, it's definitely a redeclaration and it's useful for the linter to pick it up before having to actually run the code.

**Comment by magurotuna** [comment]
OK that makes sense. I thought being compatible with ESLint / typescript-eslint would be good, but as you say it's useful even for syntax errors to be flagged when working with JS code. I'll keep this rule's behavior as it is and close this issue.
